### PR TITLE
[Blocker] Prerelease

### PR DESCRIPTION
**Pull request by guybartal**
This PR Includes:
- Checkpointing - #555 #472 
- Hyperparameters logging with mlflow #529 
- Lowercase config attributes #529 
- Config refactoring #660

### CODE REVIEW

**Comment by julia-meshcheryakova** [comment]
Wait until #660 merged. Conflict resolved there.

**Review by ritesh-modi — APPROVED** [review]
LGTM

**Pull request merged**
